DependencyInjection/SmartboxIntegrationFrameworkExtension.php

```diff
@@ -271,6 +274,55 @@ protected function loadQueueDrivers(ContainerBuilder $container)
                     }
                     break;
 
+                case 'phpamqplib':
+                    if (!class_exists(AMQPStreamConnection::class)) {
+                        throw new \RuntimeException('You need the amqp extension to use PHP-AMQP-LIB driver.');
+                    }
+
+                    if (empty($driverConfig['connections'] ?? [])) {
+                        throw new \InvalidArgumentException('You need to define at least one connection to use the AMQP driver.');
+                    }
+
+                    $driverDef = new Definition(PhpAmqpLibDriver::class);
+                    $driverDef->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
+                    $driverDef->addMethodCall('setMessageFactory', [new Reference('smartesb.message_factory')]);
+
+                    $connections = [];
+                    foreach ($driverConfig['connections'] as $index => $uri) {
+                        $connection = parse_url($uri);
+                        $connectionId = "$driverId.connection.$index";
+
+                        $connections[$index] = [
+                            'host' => $connection['host'],
+                            'port' => $connection['port'] ?? 5672,
+                            'user' => $connection['user'],
+                            'password' => $connection['pass'],
+                            'vhost' => trim($connection['path'] ?? '', '/'),
+                        ];
+                    }
+
+                    $driverDef->addMethodCall('setId', [$driverId]);
+                    $driverDef->addMethodCall('configure', [null, null, null, $driverConfig['format']]);
+                    $driverDef->addMethodCall('connect', [$connections]);
```

Comment:
Is at this point the connection stablished?

Reply:
No. The last call (to connect) is the start of the call to create the connections.

Reply:
It was modified.